jib-core/src/main/java/com/google/cloud/tools/jib/builder/BuildTarballAndLoadDockerStep.java

```diff
@@ -0,0 +1,125 @@
+    this.buildConfiguration = buildConfiguration;
+    this.listeningExecutorService = listeningExecutorService;
+    this.pullBaseImageLayerFuturesFuture = pullBaseImageLayerFuturesFuture;
+    this.buildApplicationLayerFutures = buildApplicationLayerFutures;
+    this.buildConfigurationFutureFuture = buildConfigurationFutureFuture;
+  }
+
+  /**
+   * Depends on {@code pullBaseImageLayerFuturesFuture} and {@code buildConfigurationFutureFuture}.
+   */
+  @Override
+  public Void call() throws ExecutionException, InterruptedException {
+    List<ListenableFuture<?>> dependencies = new ArrayList<>();
+    dependencies.addAll(NonBlockingFutures.get(pullBaseImageLayerFuturesFuture));
+    dependencies.addAll(buildApplicationLayerFutures);
+    dependencies.add(NonBlockingFutures.get(buildConfigurationFutureFuture));
+    return Futures.whenAllComplete(dependencies)
+        .call(this::afterPushBaseImageLayerFuturesFuture, listeningExecutorService)
+        .get();
+  }
+
+  /**
+   * Depends on {@code pushAuthorizationFuture}, {@code pushBaseImageLayerFuturesFuture.get()},
+   * {@code pushApplicationLayerFutures}, and (@code
+   * containerConfigurationBlobDescriptorFutureFuture.get()}.
+   */
+  private Void afterPushBaseImageLayerFuturesFuture()
+      throws ExecutionException, InterruptedException, IOException {
+    // Add layers to image tarball
+    TarStreamBuilder builder = new TarStreamBuilder();
+    List<String> layerFiles = new ArrayList<>();
+    for (Future<CachedLayer> cachedLayerFuture :
+        NonBlockingFutures.get(pullBaseImageLayerFuturesFuture)) {
+      Path layerFile = NonBlockingFutures.get(cachedLayerFuture).getContentFile();
+      layerFiles.add(layerFile.getFileName().toString());
+      builder.addEntry(new TarArchiveEntry(layerFile.toFile(), layerFile.getFileName().toString()));
+    }
+    for (Future<CachedLayer> cachedLayerFuture : buildApplicationLayerFutures) {
+      Path layerFile = NonBlockingFutures.get(cachedLayerFuture).getContentFile();
+      layerFiles.add(layerFile.getFileName().toString());
+      builder.addEntry(new TarArchiveEntry(layerFile.toFile(), layerFile.getFileName().toString()));
+    }
+
+    // Add config to tarball
+    Blob containerConfigurationBlob =
+        NonBlockingFutures.get(NonBlockingFutures.get(buildConfigurationFutureFuture));
+    Path tempConfig = Files.createTempFile(null, null);
+    tempConfig.toFile().deleteOnExit();
+    containerConfigurationBlob.writeTo(new BufferedOutputStream(Files.newOutputStream(tempConfig)));
+    builder.addEntry(new TarArchiveEntry(tempConfig.toFile(), "config.json"));
+
+    // Add manifest to tarball
+    ImageToJsonTranslator imageToJsonTranslator = new ImageToJsonTranslator(new Image());
+    Blob manifestBlob =
+        imageToJsonTranslator.getDockerLoadManifestBlob(
+            buildConfiguration.getTargetImageRepository(),
+            buildConfiguration.getTargetImageTag(),
```

Comment:
`BuildConfiguration` actually stores the target `ImageReference`, so could add a getter for that like `getTargetImage`. Then `ImageReference#toString` would be sufficient to get the string format

Reply:
Looks like `toString` leaves out tag if it's "latest", but we need to specify the tag in the manifest.